tracing-subscriber/src/registry/sharded.rs

```diff
@@ -0,0 +1,310 @@
+use sharded_slab::{Guard, Slab};
+
+use crate::{
+    fmt::span::SpanStack,
+    registry::{LookupSpan, SpanData},
+};
+use serde_json::{json, Value};
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    convert::TryInto,
+    fmt,
+    sync::{
+        atomic::{fence, AtomicUsize, Ordering},
+        Arc, Mutex,
+    },
+};
+use tracing_core::{
+    field::{FieldSet, Visit},
+    span::{self, Id},
+    Event, Field, Interest, Metadata, Subscriber,
+};
+
+#[derive(Debug)]
+pub struct Registry {
+    spans: Arc<Slab<Data>>,
+}
+
+thread_local! {
+    static CONTEXT: RefCell<SpanStack> = RefCell::new(SpanStack::new());
+}
+
+#[derive(Debug)]
+pub struct Data {
+    metadata: &'static Metadata<'static>,
+    parent: Option<Id>,
+    children: Vec<Id>,
+    ref_count: AtomicUsize,
+    // TODO(david): get rid of these
+    values: Mutex<HashMap<&'static str, Value>>,
+}
+
+impl Data {
```

Comment:
There's a separate `impl Data` block with other methods lower in this file...if we're going to have a `Data::drop_ref` method, it should be in that impl block with the others...

Reply:
fixed.